dev-lang/ocaml/ocaml-5.1.0.ebuild

```diff
@@ -0,0 +1,100 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit flag-o-matic
+
+HOMEPAGE="https://ocaml.org/"
+SRC_URI="https://github.com/ocaml/ocaml/archive/${PV}.tar.gz -> ${P}.tar.gz"
+DESCRIPTION="Programming language supporting functional, imperative & object-oriented styles"
+
+LICENSE="QPL-1.0 LGPL-2"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
```

Comment:
OCaml 5.x definitely doesn't work on all the arches - it's essentially a complete rewrite.

It should also go in masked.

Reply:
yes it will make problems going like that, I've found out other ml ebuilds needs updates / patches to make ecosystem to work, I've made things to work locally, opam etc... biggest confusion was created by extlib conflict, solving with `emake minimal=1 -j1 all` for extlib.

not sure what will be corect way to providing whole bunch of ebuilds, probably with things like keywords I'll need help